autopilot/contractor.go

```diff
@@ -1159,11 +1163,21 @@ func (c *contractor) formContract(ctx context.Context, w Worker, host hostdb.Hos
 	return formedContract, true, nil
 }
 
-func (c *contractor) priceTable(ctx context.Context, w Worker, hk types.PublicKey, siamuxAddr string) (*hostdb.HostPriceTable, error) {
-	ctx, cancel := context.WithTimeout(ctx, contractHostPriceTableTimeout)
+func (c *contractor) priceTable(ctx context.Context, w Worker, host hostdb.Host) (*hostdb.HostPriceTable, error) {
+	// fetch the settings if necessary
+	if host.Settings == nil {
```

Comment:
I'm not sure this fixes it in all spots. e.g. if you use the contract for uploading right after adding it.

Maybe the bus should try to scan a host when adding the contract and if that fails, create empty settings for that host in the database?

Reply:
Hm, I think the `settings` in the worker are fetched on the fly because they're part of the session right. I'm not sure whether I really like scanning the host on adding a contract because it's quite slow and shouldn't we try and keep adding contracts as slimmed down as possible to ensure we don't just add things that can prevent a contract from being inserted properly?

Something I considered though is simply defaulting to the empty settings and pricetable and just getting rid of the pointers perhaps? It will force us to change some `nil` checks but at least we won't run into nil panics but fail and error out.

```go
type Host struct {
        ...
	PriceTable   *HostPriceTable     `json:"priceTable"`
	Settings     *rhpv2.HostSettings `json:"settings"`
```
```go
	if h.Settings == (hostSettings{}) {
		hdbHost.Settings = nil
	} else {
		s := h.Settings.convert()
		hdbHost.Settings = &s
	}
```